extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -312,11 +312,11 @@ export class AmpAdExit extends AMP.BaseElement {
         this.defaultFilters_.splice(
           0,
           1,
-          createFilter(
+          /** @type {!./filters/filter.Filter} */ (createFilter(
```

Comment:
This is not a safe cast, the function explicitly can return `undefined`. Do not use a type cast to change something nullable (or that can be undefined). This can cause problems down the line. In this case, an item is being added to an array that can be `undefined`. Some piece of code later could try to use the array entry, which at that point would be `undefined`, causing an Error. The original context of it being `undefined` is lost, making the problem hard to track down.

Instead, use `devAssert(createFilter(...))`, which will change the type, but also catch the issue at this point rather than later.

Reply:
Done.

Reply:
Still not done.

Reply:
https://github.com/ampproject/amphtml/pull/23559/commits/9ba45dfc5e971f8531efb8cd7361712d6d0e5077#diff-63cf316e6647dc9b56b2d7969d48f528